### ISSUE TITLE
Childprocess v1.0.0 failing to install

### ISSUE THREAD

**Issue by fatmcgav** [enkessler/childprocess#143]
I've spotted a failure on some of our CI pipelines this morning, that appears to be related to the release of `v1.0.0` of `childprocess`.

The jobs are failing with:
```
Fetching: childprocess-1.0.0.gem (100%)
Building native extensions.  This could take a while...
ERROR:  Error installing fpm:
        ERROR: Failed to build gem native extension.

    current directory: /var/lib/gems/2.3.0/gems/childprocess-1.0.0/ext
/usr/bin/ruby2.3 mkrf_conf.rb

current directory: /var/lib/gems/2.3.0/gems/childprocess-1.0.0/ext
/usr/bin/ruby2.3 -rubygems /usr/share/rubygems-integration/all/gems/rake-10.5.0/bin/rake RUBYARCHDIR=/var/lib/gems/2.3.0/extensions/x86_64-linux/2.3.0/childprocess-1.0.0 RUBYLIBDIR=/var/lib/gems/2.3.0/extensions/x86_64-linux/2.3.0/childprocess-1.0.0
/usr/bin/ruby2.3: No such file or directory -- /usr/share/rubygems-integration/all/gems/rake-10.5.0/bin/rake (LoadError)

rake failed, exit code 1

Gem files will remain installed in /var/lib/gems/2.3.0/gems/childprocess-1.0.0 for inspection.
Results logged to /var/lib/gems/2.3.0/extensions/x86_64-linux/2.3.0/childprocess-1.0.0/gem_make.out
```

This is running on Ubuntu 16.04, however I've seen similar failures on different OS' aswell... 

Any ideas?

Cheers

**Comment by enkessler** [comment]
Hmm. I'm able to install it on both Windows 7 and Ubuntu 18. I figured that there would be breaking changes (thus the bump to 1.x) but I wasn't expecting the gem itself to actually be broken. @sds, have you noticed anything with the released gem?

@fatmcgav, for the time being, stick to the 0.x series and we'll look into it on this end. Thanks for letting us know!

**Comment by fatmcgav** [comment]
@enkessler Yeh, we've pinned to `0.9.0` for the time-being...

**Comment by farkasmate** [comment]
[fpm](https://rubygems.org/gems/fpm) is failing to install because of this.
Installing `rake` gem is fixing the issue, both `< 12.0` rake and the latest.
Should it be a runtime dependency?
https://github.com/enkessler/childprocess/blob/master/childprocess.gemspec#L24

**Comment by jordansissel** [comment]
I can't reproduce this failure on Fedora 29 -- I tested Ruby 2.3.0 and 2.6.0 (via rvm).

That said, I'll pin childprocess 0.9.0 shortly.

**Comment by farkasmate** [comment]
On it.

**Comment by enkessler** [comment]
@farkasmate I have uninstalled `rake` and I am now seeing the installation error. So, yes, the problem is that the gem requires `rake`. Now my question is 'why'? It wasn't a problem before and nothing in the `mkrf_conf.rb` file (the big new thing in this release) is jumping out at me as needing Rake. Yes, it creates a rakefile at the end but that's just a text file and isn't used. What am I missing?

**Comment by farkasmate** [comment]
@enkessler It looks like if you have an `mkrf_conf` file in extensions, than `RakeBuilder` will always run during install.
https://github.com/rubygems/rubygems/blob/511a0f5105ca95b8e389bf477b6c7cf3e90be7d1/lib/rubygems/ext/builder.rb#L125-L127

**Comment by sds** [comment]
Released [1.0.1](https://rubygems.org/gems/childprocess/versions/1.0.1) with @farkasmate's fix.
